docs/source/commandline/commandline-all.rst

```diff
@@ -346,81 +678,240 @@ OSD-related Options
     **SCRLOCK** on other operating systems (including Windows and Linux).  Use
     **FN-Delete** on Macintosh computers with notebook/compact keyboards.
 
+    Example:
+        .. code-block:: bash
+
+            mame64 ibm5150 -uimodekey DEL
+
 .. _mame-commandline-uifontprovider:
 
 **-uifontprovider**
 
-    Chooses provider for UI font rendering.
+    Chooses provider for UI font rendering. The default setting is ``auto``.
+
+.. list-table:: Supported UI font providers per-platform
+    :header-rows: 0
+    :stub-columns: 0
+
+    * - **Microsoft Windows**
+      - win
+      - dwrite
+      - none
+      - auto
+      - 
+      - sdl [#UIFPSDLWindows]_.
+    * - **macOS**
+      - 
+      - 
+      - none
+      - auto
+      - osx
+      - sdl
+    * - **Linux**
+      - 
+      - 
+      - none
+      - auto
+      - 
+      - sdl
+
+..  rubric:: Footnotes
+
+..  [#UIFPSDLWindows] SDL support on Windows requires that you compile MAME with
+                      the support in. By default SDL is not included in Windows
+                      builds of MAME.
 
-| On Windows, you can choose from: ``win``, ``dwrite``, ``none`` or ``auto``.
-| On macOS, you can choose from: ``osx``, ``none`` or ``auto``.
-| On other platforms, you can choose from: ``sdl``, ``none`` or ``auto``.
-|
-| Default setting is ``auto``.
-|
+Example:
+    .. code-block:: bash
+
+        mame64 ajax -uifontprovider dwrite
 
 .. _mame-commandline-keyboardprovider:
 
 **-keyboardprovider**
 
-    Chooses how MAME will get keyboard input.
+    Chooses how MAME will get keyboard input. The default is ``auto``.
+
+.. list-table:: Supported keyboard input providers per-platform
+    :header-rows: 0
+    :stub-columns: 0
+
+    * - **Microsoft Windows**
+      - auto [#KBIPAutoWindows]_.
+      - rawinput
+      - dinput
+      - win32
+      - none
+      - sdl [#KBIPSDLWindows]_.
+    * - **SDL (macOS and Linux)**
+      - auto [#KBIPAutoSDL]_.
+      - 
+      - 
+      - 
+      - none
+      - sdl
+    * - **Linux**
+      - auto [#KBIPAutoSDL]_.
+      - 
+      - 
+      - 
+      - none
+      - sdl
+
+..  rubric:: Footnotes
+
+..  [#KBIPAutoWindows] ``auto`` on Windows will try ``rawinput`` with
+                       fallback to ``dinput``.
+
+..  [#KBIPSDLWindows] SDL support on Windows requires that you compile MAME with
+                      the support in. By default SDL is not included in Windows
+                      builds of MAME.
+
+..  [#KBIPAutoSDL] ``auto`` on SDL will default to ``sdl``.
+
+Example:
+    .. code-block:: bash
 
-| On Windows, you can choose from: ``auto``, ``rawinput``, ``dinput``, ``win32``, or ``none``
-| On SDL, you can choose from: ``auto``, ``sdl``, ``none``
-|
-| The default is ``auto``.
-|
-| On Windows, ``auto`` will try ``rawinput`` with fallback to ``dinput``.
-| On SDL, ``auto`` will default to ``sdl``.
-|
+        mame64 c64 -keyboardprovider win32
```

Comment:
Since you’ve got a tip about joystick providers, it might be a good idea to have a tip mentioning that user-mode keyboard emulation things (e.g. Joy2Key, Remote Desktop, VLC) are more likely to work with `win32` on Windows.

Reply:
Although you really shouldn't even do that if you can at all help it because it's going to be unreliable, but yeah, that's a good one to add. It'll be on the next update.

Reply:
Since it apparently won't let me comment on the below issues directly:
Good catch on biospath, let's rip that out of the docs for the next update. I'll take another look at the anchor positioning on the other. You'll have another set within 24 hours.